src/catchup/IndexBucketsWork.h

```diff
@@ -21,6 +21,7 @@ class IndexBucketsWork : public Work
         std::shared_ptr<Bucket> mBucket;
         std::unique_ptr<BucketIndex const> mIndex;
         bool mDone{false};
+        bool mFailed{false};
```

Comment:
nit: use one BasicWork::State variable to convey final state instead of mDone and mFailed. Also it needs to be reset in doReset (right now this works because we override the retry policy, but it's a footgun)